### PR TITLE
refactor: introduce game turn

### PR DESCRIPTION
**Pull request by iandronis**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+repos:
+- repo: https://github.com/ambv/black
+  rev: 24.8.0
+  hooks:
+  - id: black
+    language_version: python3.9
```
Comment (iandronis) [inline .pre-commit-config.yaml:6]: We are using python 3.10 so we need to change this + declare somewhere the python version we are using (ex. inside pipfile).

**Review by TheraOzz — APPROVED** [review]
great job!

File: app/models/common.py [hunk 2]
```diff
@@ -1,22 +1,18 @@
...
+        from_pos_board, to_pos_board = board_move_split[0], board_move_split[1]
+        self.from_pos = self.__position_converter(from_pos_board)
+        self.to_pos = self.__position_converter(to_pos_board)
 
     @staticmethod
-    def position_converter(pos_board: str) -> tuple[int, int]:
+    def __position_converter(pos_board: str) -> tuple[int, int]:
         x_pos_board, y_pos_board = (pos_board[0].upper(), pos_board[1])
 
         x_pos_array = np.where(COLUMN_NAMES == x_pos_board)[0][0]
```
Comment (TheraOzz) [inline app/models/common.py:18]: in this method, let's return x_pos_array (array of letters) first and y_pos_array (array of numbers) second, as letters always come first in chess.
this adjustment will impact all piece move validations, but it's best to correct this now than later.

Comment (iandronis) [reply app/models/common.py]: agreed

**Pull request merged**